pyiron_base/jobs/job/generic.py

```diff
@@ -1043,8 +1048,10 @@ def to_dict(self):
         data_dict = self._type_to_dict()
         data_dict["status"] = self.status.string
         data_dict["input/generic_dict"] = {
-            "restart_file_list": self._restart_file_list,
-            "restart_file_dict": self._restart_file_dict,
+            "restart_file_list": [str(f) for f in self._restart_file_list],
+            "restart_file_dict": {
+                str(f): n for f, n in self._restart_file_dict.items()
+            },
```

Comment:
I do not think these changes are necessary, because the parameters are already converted to strings when setting the property.

Reply:
If I see the example correctly `wrap_executable` appends directly to the `restart_file_list` here

https://github.com/pyiron/pyiron_base/blob/c1aa43070dbc825788a4392785db22c4e8eb3eb8/pyiron_base/project/generic.py#L434

In the setters for `restart_file_list` I see no logic to coerce values to string as well

https://github.com/pyiron/pyiron_base/blob/c1aa43070dbc825788a4392785db22c4e8eb3eb8/pyiron_base/jobs/job/generic.py#L281